transports/quic/src/config.rs

```diff
@@ -90,6 +90,12 @@ impl Config {
         }
     }
 
+    // Set MTU discovery config
+    pub fn with_mtu_discovery_config(mut self, config: MtuDiscoveryConfig) -> Self {
```

Comment:
we should not expose `quinn`'s internal types. 
Meanwhile I checked and we already expose disabling MTU discovery: https://github.com/libp2p/rust-libp2p/. Which `MtuDiscoveryConfig` setting were you looking into expose?

Reply:
I don't see an issue with re-exporting types from other libraries, but `set_mtu_upper_bound(value: u16)` works for me. Let me change it.

Reply:
>Exposing types on other libraries, forces libp2p to make a breaking change, whenever the upstream library makes a breaking change, whether it is an actual breaking change for libp2p or not.

I could argue that when `quinn` adds or removes a an option from `MtuDiscoveryConfig`, and libp2p maintainer makes a decision to include this change, it should probably be made available to libp2p users as well. But that's definitely not a hill I want to die on, so I've modified the code to expose just `with_mtu_upper_bound`.